src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestContextInformation.cs

```diff
@@ -93,19 +93,23 @@ public void Dispose()
     }
 
     /// <summary>
-    ///     Starts a new VsTest instance and returns a wrapper to control it.
+    /// Starts a new VsTest instance and returns a wrapper to control it.
     /// </summary>
     /// <param name="runnerId">Name of the instance to create (used in log files)</param>
+    /// <param name="controlVariable">name of the env variable storing the active mutation id</param>
     /// <returns>a <see cref="IVsTestConsoleWrapper" /> controlling the created instance.</returns>
-    public IVsTestConsoleWrapper BuildVsTestWrapper(string runnerId)
+    public IVsTestConsoleWrapper BuildVsTestWrapper(string runnerId, string controlVariable)
     {
-        var vsTestConsole = _wrapperBuilder(DetermineConsoleParameters(runnerId));
+        var env = DetermineConsoleParameters(runnerId);
+        env.EnvironmentVariables["DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX"]="2";
+        // we define a per runner control variable to prevent conflict
+        env.EnvironmentVariables["STRYKER_MUTANT_ID_CONTROL_VAR"] = controlVariable;
+        var vsTestConsole = _wrapperBuilder(env);
         try
         {
             // Set roll forward on no candidate fx so vstest console can start on incompatible dotnet core runtimes
```

Comment:
```suggestion
        var env = DetermineConsoleParameters(runnerId);
        // Set roll forward on no candidate fx so vstest console can start on incompatible dotnet core runtimes
        env.EnvironmentVariables["DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX"]="2";
        // we define a per runner control variable to prevent conflict
        env.EnvironmentVariables["STRYKER_MUTANT_ID_CONTROL_VAR"] = controlVariable;
        var vsTestConsole = _wrapperBuilder(env);
        try
        {
```